diff --git hw/riscv/boot.c hw/riscv/boot.c
index cf4e5d5..c81eb63 100644
--- hw/riscv/boot.c
+++ hw/riscv/boot.c
@@ -64,6 +64,9 @@ hwaddr riscv_load_firmware(const char *filename)
     kernel_offset = (firmware_end + 0x1fffff) & ~0x1fffff;
 #endif
 
+    //Do not ever relocate kernel... (firmware is not always bbl)
+    kernel_offset = 0;
+    
     boot_debug("entry=0x" TARGET_FMT_plx " start=0x" TARGET_FMT_plx " "
                "end=0x" TARGET_FMT_plx " kernel_offset=0x" TARGET_FMT_plx "\n",
                firmware_entry, firmware_start, firmware_end, kernel_offset);
