diff --git machine/htif.c machine/htif.c
index 44ec2dd..1f0e193 100644
--- machine/htif.c
+++ machine/htif.c
@@ -125,14 +125,15 @@ static void htif_done(const struct fdt_scan_node *node, void *extra)
 
 void query_htif(uintptr_t fdt)
 {
-  struct fdt_cb cb;
-  struct htif_scan scan;
+  htif = 1;
+  //struct fdt_cb cb;
+  //struct htif_scan scan;
 
-  memset(&cb, 0, sizeof(cb));
-  cb.open = htif_open;
-  cb.prop = htif_prop;
-  cb.done = htif_done;
-  cb.extra = &scan;
+  //memset(&cb, 0, sizeof(cb));
+  //cb.open = htif_open;
+  //cb.prop = htif_prop;
+  //cb.done = htif_done;
+  //cb.extra = &scan;
 
-  fdt_scan(fdt, &cb);
+  //fdt_scan(fdt, &cb);
 }
diff --git machine/uart.c machine/uart.c
index 0645500..4494368 100644
--- machine/uart.c
+++ machine/uart.c
@@ -57,6 +57,7 @@ static void uart_done(const struct fdt_scan_node *node, void *extra)
 
   // Enable Rx/Tx channels
   uart = (void*)(uintptr_t)scan->reg;
+  uart[UART_REG_DIV] = 868;
   uart[UART_REG_TXCTRL] = UART_TXEN;
   uart[UART_REG_RXCTRL] = UART_RXEN;
 }
diff --git machine/uart.h machine/uart.h
index 51e951f..d969356 100644
--- machine/uart.h
+++ machine/uart.h
@@ -9,7 +9,7 @@ extern volatile uint32_t* uart;
 #define UART_REG_RXFIFO		1
 #define UART_REG_TXCTRL		2
 #define UART_REG_RXCTRL		3
-#define UART_REG_DIV		4
+#define UART_REG_DIV		6
 
 #define UART_TXEN		 0x1
 #define UART_RXEN		 0x1
