diff --git arch/riscv/include/uapi/asm/unistd.h arch/riscv/include/uapi/asm/unistd.h
index 13ce76cc5a..95217ebb63 100644
--- arch/riscv/include/uapi/asm/unistd.h
+++ arch/riscv/include/uapi/asm/unistd.h
@@ -17,9 +17,12 @@
 
 #ifdef __LP64__
 #define __ARCH_WANT_NEW_STAT
-#define __ARCH_WANT_SET_GET_RLIMIT
 #define __ARCH_WANT_SYS_CLONE3
 #endif /* __LP64__ */
+#define __ARCH_WANT_SET_GET_RLIMIT
+#ifndef __LP64__
+#define __ARCH_WANT_TIME32_SYSCALLS
+#endif
 
 #include <asm-generic/unistd.h>
 
diff --git arch/riscv/mm/init.c arch/riscv/mm/init.c
index 736de6c873..ad9ebf2e9a 100644
--- arch/riscv/mm/init.c
+++ arch/riscv/mm/init.c
@@ -22,6 +22,8 @@
 
 #include "../kernel/head.h"
 
+#include <linux/dma-contiguous.h>
+
 unsigned long empty_zero_page[PAGE_SIZE / sizeof(unsigned long)]
 							__page_aligned_bss;
 EXPORT_SYMBOL(empty_zero_page);
@@ -524,6 +526,7 @@ void __init paging_init(void)
 	sparse_init();
 	setup_zero_page();
 	zone_sizes_init();
+	dma_contiguous_reserve(memblock_end_of_DRAM());
 }
 
 #ifdef CONFIG_SPARSEMEM_VMEMMAP
