diff --git arch/riscv/mm/init.c arch/riscv/mm/init.c
index 42bf939693d3..928d5d80f103 100644
--- arch/riscv/mm/init.c
+++ arch/riscv/mm/init.c
@@ -18,6 +18,8 @@
 #include <asm/pgtable.h>
 #include <asm/io.h>
 
+#include <linux/dma-contiguous.h>
+
 unsigned long empty_zero_page[PAGE_SIZE / sizeof(unsigned long)]
 							__page_aligned_bss;
 EXPORT_SYMBOL(empty_zero_page);
@@ -444,4 +446,5 @@ void __init paging_init(void)
 	setup_vm_final();
 	setup_zero_page();
 	zone_sizes_init();
+	dma_contiguous_reserve(memblock_end_of_DRAM());
 }
